src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -0,0 +1,328 @@
+                            3>{{physical_x, physical_y, physical_z}}
+               : std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3>{
+                     {physical_x, -physical_y, -physical_z}};
+  }
+  // Wedges on y axis:
+  if (positioning_of_wedge_.axis() == Direction<3>::Axis::Eta) {
+    return (positioning_of_wedge_.side() == Side::Upper)
+               ? std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+                            3>{{physical_y, physical_z, physical_x}}
+               : std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3>{
+                     {physical_y, -physical_z, -physical_x}};
+  }
+  // Wedges on x axis:
+  if (positioning_of_wedge_.axis() == Direction<3>::Axis::Xi) {
+    return (positioning_of_wedge_.side() == Side::Upper)
+               ? std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+                            3>{{physical_z, physical_x, physical_y}}
+               : std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3>{
+                     {-physical_z, -physical_x, physical_y}};
+  }
+  ERROR("Improper axis passed to function.");
+}
+
+// Currently unimplemented, hence the noreturn attribute
+template <typename T>
+[[noreturn]] std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3>
+Wedge3D::inverse(const std::array<T, 3>& /*x*/) const {
+  ERROR("Inverse map is unimplemented for Wedge3D");
+}
+
+template <typename T>
+Tensor<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+       tmpl::integral_list<std::int32_t, 2, 1>,
+       index_list<SpatialIndex<3, UpLo::Up, Frame::NoFrame>,
+                  SpatialIndex<3, UpLo::Lo, Frame::NoFrame>>>
+Wedge3D::jacobian(const std::array<T, 3>& x) const {
+  using ReturnType = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+
+  const ReturnType xi = x[0];
+  const ReturnType eta = x[1];
+  const ReturnType zeta = x[2];
+  const double ratio_of_radii = outer_radius_ / inner_radius_;
+  const ReturnType x_equiangular_secant =
+      make_with_value<ReturnType>(xi, 1.0) / cos(M_PI_4 * xi);
+  const ReturnType y_equiangular_secant =
+      make_with_value<ReturnType>(xi, 1.0) / cos(M_PI_4 * eta);
+  const ReturnType x_equiangular_tangent = tan(M_PI_4 * xi);
+  const ReturnType y_equiangular_tangent = tan(M_PI_4 * eta);
+
+  const ReturnType sy_square = square(y_equiangular_secant);
+  const ReturnType sx_square = square(x_equiangular_secant);
+  const ReturnType scaling_and_sphericity =
+      M_PI * sqrt(3.0) *
+      (ratio_of_radii * (1 + zeta) + inner_sphericity_ * (1 - zeta));
+  const ReturnType denominator_dzeta =
+      (2.0 * sqrt(sx_square + square(y_equiangular_tangent))) / sqrt(3.0);
+  const ReturnType denominator_other =
+      8.0 * pow<3>(0.5 * sqrt(3.0) * denominator_dzeta);
+  const ReturnType nonsphericity =
+      (-1.0 + inner_sphericity_) * 0.5 * denominator_dzeta;
+  const ReturnType trigonometric_nonsphericity =
```

Comment:
This is the best variable name I could come up with, but I don't think it is a very good name